packages/cli/bin/protosaurus.js

```diff
@@ -19,70 +19,90 @@
 // https://github.com/sindresorhus/meow/issues/69 is fixed (regarding subcommands).
-//
-// Clean cache:
-// $ protosaurus clean
-//
-// Generate resources:
-// $ protosaurus generate
+const AVAILABLE_COMMANDS = ['start', 'build', 'clean', 'generate'];
 
 (async () => {
-  const [_node, _protosaurus, command, relativePathToBufGenYaml] = process.argv;
-  if (command !== 'clean' && command !== 'generate') {
+  const [_node, _protosaurus, command, relativePathArgs] = process.argv;
+  if (!AVAILABLE_COMMANDS.includes(command)) {
+    const commandsString = AVAILABLE_COMMANDS.map((cmd) => `\`${cmd}\``).join(
+      ', '
+    );
+
     throw new Error(
-      `Invalid command ${command}. Currently available ones are \`clean\` and \`generate\`.`
+      `Invalid command ${command}. Currently available ones are ${commandsString}.`
     );
   }
 
   const fs = await import('fs-extra');
+  const relativePathToBufGenYaml = relativePathArgs || '../';
+  const pathToBufGenYaml = path.join(WORK_DIR, relativePathToBufGenYaml);
 
   switch (command) {
     case 'clean': {
-      fs.rmdir(path.join(DOCUSAURUS_DIR, '.protosaurus'));
+      fs.rmdir(path.join(WORK_DIR, '.protosaurus'));
+      break;
     }
     case 'generate': {
-      const pathToBufGenYaml = path.join(
-        DOCUSAURUS_DIR,
-        relativePathToBufGenYaml
-      );
-
-      // TODO(imballinst): identify cache by content.
-      const currentListOfFiles = await generator.getListOfProtoFiles({
-        workDir: relativePathToBufGenYaml
-      });
-
-      const { pathToCache } = mdx.getPathsToCache(DOCUSAURUS_DIR);
-      const isCacheInvalid = await mdx.isCacheInvalid({
-        pathToCache,
-        currentListOfFiles
-      });
-
-      // Check cache status, then determine whether MDX/JSON dictionary need to be
-      // emitted or not.
-      if (isCacheInvalid) {
-        console.info(
-          'There were one or more `.proto` files changed since last build, regenerating...'
+      await generate(pathToBufGenYaml);
+      break;
+    }
+    case 'start':
+    case 'build': {
+      if (relativePathArgs === undefined) {
+        console.warn(
+          "No relative path argument to buf.gen.yaml directory given. It will default to '../'."
         );
-        // Generate protoc JSON.
-        await generator.generate({
-          workDir: pathToBufGenYaml,
-          outPath: `${DOCUSAURUS_DIR}/.protosaurus/generated`
-        });
-        // Generate MDX and JSON dictionary from the generated JSON above.
-        await mdx.emitJsonAndMdx(DOCUSAURUS_DIR);
       }
 
-      await generator.generateCacheFile({
-        outPath: `${DOCUSAURUS_DIR}/.protosaurus/plugin-resources/.cache`,
-        newList: currentListOfFiles
+      // When starting the dev server or building, generate the
+      // files first.
+      await generate(pathToBufGenYaml);
+      // After that, run docusaurus.
+      spawn('yarn', ['docusaurus', command], {
```

Comment:
This is not awaited, right? Though `execa` has its flaws, probably it is not bad to use that instead.

Reply:
Ah, yes, we can use `execa`. I'll update it tomorrow. Thanks for the suggestion!

But I'm curious about its purpose, is it only for making the package uniform (with async)? Or is there a particular benefit with spawning child process asynchronously?

Reply:
Yeah, we can call sync here that's fine, the reason is to capture the error correctly. Running in nextTick for `spawn` probably will have a problem since we don't have the "handle" for that.

Reply:
Got it. Thanks for the explanation!